### PR TITLE
Fixed some bugs and added unit tests

### PR DESCRIPTION
**Pull request by Hindsight420**

### CODE REVIEW

File: Assets/_Scripts/2Models/Tile.cs [hunk 1]
```diff
@@ -49,6 +49,11 @@ public override string ToString()
         {
             return $"Tile: {X}, {Y}";
         }
-    }
 
+        public override bool Equals(object obj)
```
Comment (JackRiga) [inline Assets/_Scripts/2Models/Tile.cs]: Make this an operator method

Comment (JackRiga) [reply Assets/_Scripts/2Models/Tile.cs]: public static Calculator operator + (Calculator Calc1, Calculator Calc2)

Comment (Hindsight420) [reply Assets/_Scripts/2Models/Tile.cs]: Done

Comment (Hindsight420) [reply Assets/_Scripts/2Models/Tile.cs]: Nvm, I don't know why, but adding those operator overrides makes the test fail.

Comment (Hindsight420) [reply Assets/_Scripts/2Models/Tile.cs]: I found a fix and pushed it. Still a bug that makes the TestAdjacentTilesBorders run red, will investigate later.

**Review by JackRiga — APPROVED** [review]

**Pull request merged**